io-hdfs/src/main/java/cz/o2/proxima/storage/hdfs/HdfsDataAccessor.java

```diff
@@ -77,24 +77,25 @@
   private static final long HDFS_BATCH_PROCES_SIZE_MIN_DEFAULT = 1024 * 1024 * 100; /* 100 MiB */
 
   private static final Pattern PART_FILE_PARSER = Pattern.compile("part-([0-9]+)_([0-9]+)-.+");
+  private static final DateTimeFormatter DIR_FORMAT = DateTimeFormatter.ofPattern("/yyyy/MM/");
 
-  private final Configuration conf;
-  private final FileSystem fs;
+  private final Map<String, Object> cfg;
+
+  private transient Configuration conf;
+  private transient FileSystem fs;
   private final int minElementsToFlush;
   private final long rollInterval;
   private final long batchProcessSize;
 
-  private final DateTimeFormatter DIR_FORMAT = DateTimeFormatter.ofPattern("/yyyy/MM/");
-
-  private SequenceFile.Writer writer = null;
-  private Path writerTmpPath = null;
+  private transient SequenceFile.Writer writer = null;
+  private transient Path writerTmpPath = null;
   private long lastRoll = 0;
   private long elementsSinceFlush = 0;
   private long minElementStamp = Long.MAX_VALUE;
   private long maxElementStamp = Long.MIN_VALUE;
   private long monothonicTime = 0L;
 
-  private Executor executor;
+  private transient Executor executor;
```

Comment:
Hm, this would have to be done in a little more complex way. The executor is created from serializable `Context`. We must split his file into two classes - one implementing the `DataAcessor` and extending `AbstractBulkAttributeWriter` and one implementing `BatchLogObservable`. Then we must pass the `Context` from `getBatchLogObservable` to the newly created class and then create the executor using the context when it is null and store the instance.

Reply:
Done